plugins/inputs/uwsgi/uwsgi.go

```
@@ -0,0 +1,181 @@
+package uwsgi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+var timeout = 5 * time.Second
+
+type Uwsgi struct {
+	Servers []string `toml:"server"`
+}
+
+func (u *Uwsgi) Description() string {
+	return "Read uWSGI metrics."
+}
+
+func (u *Uwsgi) SampleConfig() string {
+	return `
+    ## List with urls of uWSGI Stats servers. Url must match pattern:
+    ## scheme://address[:port]
+    ##
+    ## For example:
+    ## servers = ["tcp://localhost:5050", "http://localhost:1717", "unix:///tmp/statsock"]
+    servers = []
+`
+}
+
+func (u *Uwsgi) Gather(acc telegraf.Accumulator) error {
+	for _, s := range u.Servers {
+		n, err := url.Parse(s)
+		if err != nil {
+			return fmt.Errorf("Could not parse uWSGI Stats Server url '%s': %s", s, err)
+		}
+
+		u.gatherServer(acc, n)
+
+	}
+	return nil
+}
+
+func (u *Uwsgi) gatherServer(acc telegraf.Accumulator, url *url.URL) error {
+	var err error
+	var r io.ReadCloser
+
+	switch url.Scheme {
+	case "unix":
+		r, err = net.DialTimeout(url.Scheme, url.Path, timeout)
+	case "tcp":
+		r, err = net.DialTimeout(url.Scheme, url.Host, timeout)
+	case "http":
+		resp, err := http.Get(url.String())
+		if err != nil {
+			return fmt.Errorf("Could not connect to uWSGI Stats Server '%s': %s", url.String(), err)
+		}
+		r = resp.Body
+	default:
+		return fmt.Errorf("'%s' is not a valid URL", url.String())
+	}
+
+	if err != nil {
+		return fmt.Errorf("Could not connect to uWSGI Stats Server '%s': %s", url.String(), err)
+	}
+	defer r.Close()
+
+	var s StatsServer
+	s.Url = url.String()
+
+	dec := json.NewDecoder(r)
+	dec.Decode(&s)
```

don't ignore errors here either